File: API/Helpers/UserParams.cs
namespace API.Helpers
{

    // We inherit from the PaginationParams class so that we still get acess to the PaginationParams properties
    public class UserParams : PaginationParams
    {
        public string CurrentUserName { get; set; }

        public string Gender { get; set; }

        public int MinAge { get; set; } = 18;

        public int MaxAge { get; set; } = 150;

        public string OrderBy { get; set; } = "lastActive"; 


    }
}

File: client/src/app/Models/userParams.ts
import { User } from "./user";

export class UserParams { // This class will be passed into our getMembers method in our member service file
    gender: string;
    minAge = 18;
    maxAge = 99;
    pageNumber = 1;
    pageSize = 5;
    orderBy = 'lasActive'; // This will enable the order of the users that is returned from our htp request will be by the lastActive users property

    constructor (user: User) {
        this.gender = user.gender === 'female' ? 'male' : 'female';
    }
}